packages/framework/src/Model/Order/OrderData.php

```
@@ -310,10 +310,23 @@ public function addTotalPrice(PriceInterface $priceToAdd, string $type): void
 
     /**
      * @param \Shopsys\FrameworkBundle\Model\Order\Item\OrderItemData $item
+     * @param bool $new
      */
-    public function addItem(OrderItemData $item): void
+    public function addItem(OrderItemData $item, bool $new = false): void
```

I'm not sure about this. It brings much confusion. 
The addItem() method is called in AddXXX middlewares, and there is always created a new item, but the method is called without setting $new=true.

Also, how should it behave when I call consecutively add with the same product, once with add, once without?

```
$orderData->addItem($orderItemData1, true);
$orderData->addItem($orderItemData1, false);
```

Now, the item is silently ignored. Which tells me that for some cases I must add $new=true (calling productFacade::edit()), while for the others I don't (middlewares). 

I would rather check the failing scenario during productfacade::edit and try to fix it there.

---

I have reworked it